src/main/java/org/apache/sling/servlets/post/JSONResponse.java

```diff
@@ -103,9 +104,13 @@ JsonObject getJson() {
         JsonObjectBuilder jsonBuilder = Json.createObjectBuilder();
         for (Map.Entry<String, Object> entry : json.entrySet()) {
             if (entry.getValue() != null) {
-                jsonBuilder.add(entry.getKey(), entry.getValue().toString());
-            }
-            else {
+                try {
```

Comment:
Instead of relying on exception handling please rather use an explicit check: https://docs.oracle.com/javaee/7/api/javax/json/JsonValue.html#getValueType--

Reply:
@kwin  - That's difficult because the class type of entry value can't be restricted to a range. We can't really exhaustively give all the possible json class names.

Reply:
@kwin - That's difficult because the class type of entry value is not always javax json. If I am not too wrong, it can as well be org.apache.sling.commons.json.JSONObject or com.google.gson.JsonObject. We can't really restrict the end user to only javax json or can we?

Reply:
I think instead of buffering the values in a map, we should directly use the according add methods in https://docs.oracle.com/javaee/7/api/javax/json/JsonObjectBuilder.html and throw an exception if an invalid type is passed to `setProperty`. That way we create the JSON Object on the fly and make sure that we don't accept any invalid value types.

Reply:
@kwin - But even than the issue remains-neither we know the "type", nor we can exhaustively check all "types"